device_calendar/ios/Classes/SwiftDeviceCalendarPlugin.swift

```diff
@@ -159,6 +171,33 @@ public class SwiftDeviceCalendarPlugin: NSObject, FlutterPlugin {
             }
             
         }
+        var recurrenceRule: RecurrenceRule?
+        if(ekEvent.hasRecurrenceRules) {
+            let ekRecurrenceRule = ekEvent.recurrenceRules![0]
+            var frequency: Int64
+            switch ekRecurrenceRule.frequency {
+            case EKRecurrenceFrequency.daily:
+                frequency = 0
+            case EKRecurrenceFrequency.weekly:
+                frequency = 1
+            case EKRecurrenceFrequency.monthly:
+                frequency = 2
+            default:
+                frequency = 0
+            }
+            
+            var totalOccurrences: Int? = nil
+            var endDate: Int64? = nil
+            if((ekRecurrenceRule.recurrenceEnd?.occurrenceCount) != nil) {
```

Comment:
Is the extra parentheses required here?

Reply:
was inserted by XCode when i applied a fix. will remove as it's optional